include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -185,29 +185,82 @@ struct __init_processing
 //------------------------------------------------------------------------
 
 template <typename _ExecutionPolicy, typename _Operation1, typename _Operation2>
-struct transform_init
+struct transform_init_seq
 {
     _Operation1 __binary_op;
     _Operation2 __unary_op;
 
-    template <typename _NDItemId, typename _Size, typename _AccLocal, typename... _Acc>
+    template <typename _Size, typename _AccLocal, typename... _Acc>
     void
-    operator()(const _NDItemId __item, _Size __n, ::std::size_t __iters_per_work_item, ::std::size_t __global_id,
-               ::std::size_t __global_offset, _AccLocal& __local_mem, const _Acc&... __acc) const
+    operator()(const _Size __n, _AccLocal& __local_mem, const _Acc&... __acc) const
+    {
+        __local_mem = __unary_op(0, __acc...);
+        // Add neighbour to the current __local_mem
+        for (_Size __i = 1; __i < __n; ++__i)
+            __local_mem = __binary_op(__local_mem, __unary_op(__i, __acc...));
```

Comment:
Just an idea, regarding an unary transformation:
I have been keeping in mind long time an idea about usage of a kind of lazy range transformation (transfor_view or oneapi::dpl::__ranges::transform_view_simple) instead of explicit code for unary operation processed.
(https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/utils_ranges.h#L312)

Reply:
This would apply to several patterns, right? Should we address this then in a separate PR?

Reply:
Probably, that may be impossible to apply to the all algo patterns, which use the reduce SYCL pattern... But I hope.
It is a kind of refactoring. It may be done sometime in the future. Probably it may require additional experiments... It is not urgent I think.